epitator/infection_annotator.py

```diff
@@ -262,9 +260,70 @@ def from_noun_chunks_with_infection_lemmas(doc, debug=False):
     return(infection_spans)
 
 
+def collapse_span_group(span_group):
+    if len(span_group.base_spans) == 0:
+        # Not a span group
+        return span_group
+    collapsed_metadata = {
+        'attributes': []
+    }
+    if span_group.label:
+        collapsed_metadata['attributes'] = [span_group.label]
+    for span in span_group.base_spans:
+        span = collapse_span_group(span)
+        if span.metadata:
+            collapsed_metadata = dict(
+                dict(collapsed_metadata, **span.metadata),
+                attributes=collapsed_metadata['attributes'] + span.metadata['attributes'])
```

Comment:
I wrote code somewhere to manage collapsing dictionaries — perhaps it can be used here. I'll look it out in a second.